### PR TITLE
REL-2593: updating connector-py release notes for 6.1.2.21

### PR DESCRIPTION
**Pull request by rkorte**
Here is the documentation output associated with this PR: https://jenkins-community.rti.com/job/ci/job/Connector-PY/job/rticonnextdds-connector-py-doc/view/change-requests/job/PR-189/Connector_20Documentation/release_notes.html

### CODE REVIEW

**Comment by rkorte** [comment]
@Alxe, do we need to add return_samples to the documentation, perhaps [here](https://jenkins-community.rti.com/job/ci/job/Connector-PY/job/rticonnextdds-connector-py-doc/job/release%252Fconnector%252F1.2.4/Connector_20Documentation/input.html#input-class)?

Can you update the documentation in this PR, and I will review it?

File: docs/release_notes.rst [hunk 1]
```diff
@@ -32,15 +32,42 @@ runs on most Windows®, Linux® and macOS® platforms.
...
+Using the ``wait`` operation of an ``Input`` after doing ``read`` or ``take``
+could have led to a deadlock scenario. This only happened if ``read`` or
+``take`` returned the the maximum amount of samples that an ``Input`` can hold.
+
+To remedy this, a new ``return_samples`` operation has been added to Inputs.
+Additionally, the ``wait`` operation can take a ``return_samples=True`` keyword
+argument to trigger the functionality. This behavior is not enabled by default,
+to be backwards compatible.
+
+.. code:: python
```
Comment (rkorte) [inline docs/release_notes.rst:55]: @Alxe, correct code argument?

File: docs/copyright_license.rst
Comment (rkorte) [inline docs/copyright_license.rst:1]: The [json-parser](https://jenkins-community.rti.com/job/ci/job/Connector-PY/job/rticonnextdds-connector-py-doc/job/release%252Fconnector%252F1.2.4/Connector_20Documentation/copyright_license.html) entry in this file does not exactly match the [3rdParty document in the Pro installation](https://docserver.rti.com/docs/connext-docs/6.1.2.21/doc/manuals/connext_dds_professional/release_notes_3rdparty/RTI_ConnextDDS_3rdPartySoftware_CoreLibraries.pdf). Which one should I update to match the other?

**Comment by Alxe** [comment]
> @Alxe, do we need to add return_samples to the documentation, perhaps [here](https://jenkins-community.rti.com/job/ci/job/Connector-PY/job/rticonnextdds-connector-py-doc/job/release%252Fconnector%252F1.2.4/Connector_20Documentation/input.html#input-class)?
> 
> Can you update the documentation in this PR, and I will review it?

I think we already have that in the documentation?
image

File: docs/release_notes.rst (hunk 1 above)
Comment (Alxe) [reply docs/release_notes.rst:55]: Yes, that'll be alright. :)

File: docs/copyright_license.rst
Comment (Alxe) [reply docs/copyright_license.rst:1]: `https://github.com/udp/json-parser` (Connector documentation) redirects to `https://github.com/json-parser/json-parser` (3rd Party document), so I'd say we need to update this section.

**Review by Alxe — CHANGES_REQUESTED** [review]
Update 3rd Party notes.

File: docs/copyright_license.rst
Comment (rkorte) [reply docs/copyright_license.rst:1]: @Alxe, I updated the (c) line, too, to match what we have on https://github.com/json-parser/json-parser

**Review by Alxe — APPROVED** [review]

**Pull request merged**